Traceback (most recent call last):
  File "/Users/isaacflath/opt/miniconda3/envs/p38/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/Users/isaacflath/opt/miniconda3/envs/p38/lib/python3.8/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/isaacflath/opt/miniconda3/envs/p38/lib/python3.8/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/isaacflath/opt/miniconda3/envs/p38/lib/python3.8/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/Users/isaacflath/opt/miniconda3/envs/p38/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/Users/isaacflath/opt/miniconda3/envs/p38/lib/python3.8/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/Users/isaacflath/opt/miniconda3/envs/p38/lib/python3.8/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/isaacflath/opt/miniconda3/envs/p38/lib/python3.8/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from sklearn.linear_model import LinearRegression

fig,ax = plt.subplots(figsize=(15,8))
_ = df['AAPL'].iloc[:10]
x1 = _.index.day.values
y1 = _.values

sz = 4
regr_y = [y1[i:i+sz] for i in range(0,len(y1)-(sz-1))]
regr_x = [x1[i:i+sz].reshape(-1, 1) for i in range(0,len(x1)-(sz-1))]

regr = [LinearRegression().fit(x,y).predict(x) for x,y in zip(regr_x,regr_y)]


regr_padded = [[None]*i+list(r)+[None]*(len(x)-sz-i) for i,r in enumerate(regr)]


ax.scatter(_.index,y1)
for i in range(len(regr)): ax.plot(_.index,regr_padded[i])

fig.suptitle("Window Regression Lines",fontsize=50)

------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Input [0;32mIn [12][0m, in [0;36m<cell line: 1>[0;34m()[0m
[0;32m----> 1[0m [38;5;28;01mfrom[39;00m [38;5;21;01msklearn[39;00m[38;5;21;01m.[39;00m[38;5;21;01mlinear_model[39;00m [38;5;28;01mimport[39;00m LinearRegression
[1;32m      3[0m fig,ax [38;5;241m=[39m plt[38;5;241m.[39msubplots(figsize[38;5;241m=[39m([38;5;241m15[39m,[38;5;241m8[39m))
[1;32m      4[0m _ [38;5;241m=[39m df[[38;5;124m'[39m[38;5;124mAAPL[39m[38;5;124m'[39m][38;5;241m.[39miloc[:[38;5;241m10[39m]

[0;31mModuleNotFoundError[0m: No module named 'sklearn'
ModuleNotFoundError: No module named 'sklearn'

